src/filter/http_context.rs

```diff
@@ -110,19 +123,29 @@ impl Context for Filter {
             self.context_id
         );
 
-        let some_op = self.operation_dispatcher.borrow().get_operation(token_id);
+        let op_res = self
+            .operation_dispatcher
+            .borrow()
+            .get_waiting_operation(token_id);
 
-        if let Some(operation) = some_op {
-            if GrpcService::process_grpc_response(operation, resp_size).is_ok() {
-                self.operation_dispatcher.borrow_mut().next();
-                if let Some(_op) = self.operation_dispatcher.borrow_mut().next() {
-                } else {
-                    self.resume_http_request()
+        match op_res {
+            Ok(operation) => {
+                if GrpcService::process_grpc_response(operation, resp_size).is_ok() {
+                    // call the next op
+                    match self.operation_dispatcher.borrow_mut().next() {
+                        Ok(_) => {} // no action needed
+                        Err(op_err) => {
+                            // If desired, we could check the error status.
+                            GrpcService::handle_error_on_grpc_response(op_err.failure_mode);
```

Comment:
Here we handle error and then we resume http request with ` self.resume_http_request();`

I wonder if that is correct.

Reply:
Actually, the flow should be:

1. On error:
  a. FailureMode::Deny => return 500
  b FailureMode::Allow => resume request
  
2. On op => nothing, already called the next op
3. On None => No more ops, resume request

will push another commit. Thanks!